ios/Wrappers/DecodedMessageWrapper.swift

```diff
@@ -13,6 +13,7 @@ struct DecodedMessageWrapper {
 			"senderAddress": model.senderAddress,
 			"sent": UInt64(model.sentAt.timeIntervalSince1970 * 1000),
 			"fallback": model.encodedContent.fallback,
+			"shouldPush": model.shouldPush
```

Comment:
@kele-leanes I think here instead of calling `model.shouldPush` maybe you can copy how the test does it

```
		let message = try await MessageV2.encode(
			client: client,
			content: model.encodedContentt,
			topic: model.topic,
			keyMaterial: Data(model.keyMaterial),
			codec: model.encodedContent.contentType // MAP THIS TO A CODEC?
		)
```

then it would be `message.shouldPush`

Or something along these lines to get the push without having to make changes to the iOS sdk directly.

Reply:
To do that I need to make `MessageV2` and `codecRegistry` public and add keyMaterial to model which is not currently sent

Reply:
and making this method `async/await` will add more complexity where this method is called

Reply:
I have no issue with `MessageV2` and `codecRegistry` being public. @nakajima do you have any thoughts?

Reply:
besides that, since the codec registry in RN lives on the JS side, the `MessageV2.encode` method can not access the codecs from RN.
e.g: If I register a custom codec, which returns `true` from the shouldPush method, the native side can’t get this flag.
Is the reason why I suggest sending the value encoded as in this [PR](https://github.com/xmtp/xmtp-ios/pull/238) or sending it as a part of a `SendOption` object

Reply:
Well we can encode it like that unless we update the proto code in https://github.com/xmtp/proto. Maybe @richardhuaaa has some ideas of how we can get around this complexity in react native.

Reply:
Just catching up here, forgive me if I misunderstand some React Native/iOS things. Are we trying to set `shouldPush` on message send here?

It is important that `should_push` is not encrypted, and presented in plaintext alongside the ciphertext: https://github.com/xmtp/proto/blob/beta/proto/message_contents/message.proto#L51

It looks like it is being set in iOS here: https://github.com/xmtp/xmtp-ios/blob/5b40ee96d4165355ef9bf0843f06737e2ee31407/Sources/XMTPiOS/Messages/MessageV2.swift#L117

Is React Native eventually calling into iOS there - if so, can we confirm iOS is already setting it for us?

Reply:
Regarding the behavior in iOS, `shouldPush` is already established and calculated based on the `codedTypes` existing on the iOS side. However, in React Native (RN), `codedTypes` are generated and stored on the JavaScript side. Consequently, the determination of this flag should occur on the JS side and then communicated to iOS. I've devised two solutions:

1. Include the `shouldPush` value as part of the `EncodedContent` using the proto, similar to how we currently send the [fallback](https://github.com/xmtp/xmtp-react-native/blob/4a4d20fe65c464959be62eda875536bfcbcfad2f/src/index.ts#L242). This value can then be decoded and set when a `MessageV2` is created. ([Link to PR](https://github.com/xmtp/xmtp-ios/pull/238))
   
2. Send the `shouldPush` value as a plain Boolean, as proposed in this PR, and include it as an optional part of the `SendOption`.

The latter approach deviates from the convention of maintaining methods as similar as possible across platforms.

Reply:
Gotcha! Thanks for explaining.

So I don't think we should do (1) - it muddies the protocol, and the protos are shared across all of the platforms as well as sent over the network.

> in React Native (RN), CodecTypes are generated and stored on the JavaScript side

More of a long-term question/not blocking this PR, but why is this the case? Is there a reason we can't push that down to the native platform?

> The latter approach deviates from the convention of maintaining methods as similar as possible across platforms.

So the issue is that we are deriving `shouldPush` from the non-encoded content, however the SDK is structured so that only the encoded content is sent down today, which means iOS is unable to derive shouldPush? If so, then does that mean RN Android has the same issue and would also benefit from this?

It seems like we have no choice with the current design but to do the `sendOption` thing even if it's a little ugly. Another question - given the `contentType` is [in the `SendOptions` already](https://github.com/xmtp/xmtp-ios?tab=readme-ov-file#handle-different-content-types), would it be feasible to have iOS construct the codec based on the content type, and then determine the `shouldPush` value from that? That way we are not directly giving app developers a way to set `shouldPush` directly.

Reply:
I see. In that case it seems that adding `shouldPush` to the `SendOptions` is the only way. Perhaps we can name it something that makes it clear that it is internal-use only, like `__shouldPush` or something.

Long-term, there will probably be more cases like this where the RN SDK wants some functionality from the iOS SDK that we don't actually want to expose publicly. I wonder if it makes sense to define an iOS type/interface that are internal-use only, that we don't document or provide any guarantees about externally.

Reply:
resolved in [ccceabe](https://github.com/xmtp/xmtp-react-native/pull/235/commits/ccceabe988f0114cd3914de428b0cc459ad83d90)